pypesto/optimize/optimizer.py

```diff
@@ -159,55 +198,56 @@ def wrapped_minimize(self, problem, x0, id, allow_failed_starts,
     return wrapped_minimize
 
 
-def fill_result_from_objective_history(
-        result: OptimizerResult,
-        optimizer_history: OptimizerHistory):
-    """Overwrite values in the result object with values in the history."""
-    update_vals = True
-    # check history for better values
-    # value could be different e.g. if constraints violated
-    if optimizer_history.fval_min is not None and result.fval is not None \
-            and not np.isclose(optimizer_history.fval_min, result.fval):
-        logger.warning(
-            "Function values from history and optimizer do not match: "
-            f"{optimizer_history.fval_min}, {result.fval}")
-        # do not use value from history
-        update_vals = False
-
-    if optimizer_history.x_min is not None and result.x is not None \
-            and not np.allclose(result.x, optimizer_history.x_min):
-        logger.warning(
-            "Parameters obtained from history and optimizer do not match: "
-            f"{optimizer_history.x_min}, {result.x}")
-        # do not use value from history
-        update_vals = result.fval is None
-
-    # update optimal values from history if reported function values
-    # and parameters coincide
-    if update_vals:
-        # override values from history if available
-        result.x = optimizer_history.x_min
-        result.fval = optimizer_history.fval_min
-        if optimizer_history.grad_min is not None:
-            result.grad = optimizer_history.grad_min
-        if optimizer_history.hess_min is not None:
-            result.hess = optimizer_history.hess_min
-        if optimizer_history.res_min is not None:
-            result.res = optimizer_history.res_min
-        if optimizer_history.sres_min is not None:
-            result.sres = optimizer_history.sres_min
+def fill_result_from_history(
+    result: OptimizerResult,
+    optimizer_history: OptimizerHistory,
+    optimize_options: OptimizeOptions = None,
+) -> OptimizerResult:
+    """Overwrite some values in the result object with values in the history.
+
+    Parameters
+    ----------
+    result: Result as reported from the used optimizer.
+    optimizer_history: History of function values recorded by the objective.
+    optimize_options: Options on e.g. how to override.
+
+    Returns
+    -------
+    result: The in-place modified result.
+    """
+    if optimize_options is None:
+        optimize_options = OptimizeOptions()
+
+    # logging
+    history_fval, result_fval = optimizer_history.fval_min, result.fval
```

Comment:
no more warning if parameters do not match?

Reply:
will add (as logging too, right?)